source/server/exceptions.h

```diff
@@ -1,14 +1,29 @@
 #ifndef NIDEVICE_GRPC_EXCEPTIONS
 #define NIDEVICE_GRPC_EXCEPTIONS
 
+#include <grpcpp/grpcpp.h>
+
 #include <stdexcept>
 
 namespace nidevice_grpc {
 
-struct ValueOutOfRangeException : std::runtime_error {
-  ValueOutOfRangeException(const char* message) : std::runtime_error(message) {}
-  ValueOutOfRangeException(const std::string& message) : std::runtime_error(message) {}
-  ValueOutOfRangeException(const ValueOutOfRangeException& other) : std::runtime_error(other) {}
+class NonDriverException : std::runtime_error {
```

Comment:
Changing `struct` to `class` also changed from public inheritance to private inheritance. Can `catch` blocks match against private base classes? https://en.cppreference.com/w/cpp/language/try_catch seems to say they only match against public base classes.

Reply:
And of course that was the reason for build failures on Unix (not sure why it didn't have a problem on Windows, but oh well):
```
/home/runner/work/grpc-device/grpc-device/source/tests/unit/session_resource_repository_tests.cpp:280:58: error: ‘virtual const char* std::runtime_error::what() const’ is inaccessible within this context
           EXPECT_STREQ(expected_message.c_str(), ex.what());
```